src/dune_rules/blang.ml

```diff
@@ -57,6 +57,96 @@ let rec eval t ~dir ~f =
     and y = String_with_vars.expand y ~mode:Many ~dir ~f in
     Op.eval op (Value.L.compare_vals ~dir x y)
 
+module Partial = struct
+  type partial_value =
+    | Unexpanded of String_with_vars.t
+    | Expanded of Value.t list
+
+  type t =
+    | Const of bool
+    | Expr of String_with_vars.t
+    | And of t list
+    | Or of t list
+    | Compare of Op.t * partial_value * partial_value
+
+  let of_bool b = Const b
+
+  let to_bool = function
+    | Const b -> Some b
+    | _ -> None
+
+  let rec eval t ~dir ~f =
+    match t with
+    | Const x -> x
+    | Expr sw -> (
```

Comment:
I just got confused here: `sw` presumably stands for `String_with_vars` but the latter has got no `w`...

Reply:
Oh, never mind, it does have a `w` -- in "with". I can't read :)